File: Assignment4/q1.java
// Written by Dallas Hart on 2023-02-01
// This program belongs to Assignment 4 in ENGG*1420 at UOG
// Purpose: Use RLE to encode a user inputted string

import java.util.Scanner;

class Question1_Assign4
{
    public static void main(String[] args)
    {
        // Grab user input
        Scanner scanner = new Scanner(System.in);
        String input = scanner.nextLine();

        // Run length encoding
        char last = '\0'; // Placeholder for first element
        int count = 1;
        for (int i = 0; i < input.length(); i++) {
            if (input.charAt(i) == last){ // Check if element is the same as the last
                count++;
            } else if (last != '\0') { // Only print for non placeholder elements
                // Print character/length for RLE
                System.out.print(last);
                if (count != 1) { // skip count if there is only 1.
                    System.out.print(count);
                    count = 1;
                }
            }
            last = input.charAt(i); // Update last element
        }
        // Print out last character/length combo
        System.out.print(last);
        if (count != 1) {
            System.out.print(count);
            count = 1;
        }

        // Add ending newline and close scanner
        System.out.println("");
        scanner.close();
    }
}

File: ENGG-1420/Assignment1/q1.c
// Written by Dallas Hart on 2023-01-11
// This program belongs to Assignment 1 in ENGG*1420 at UOG
// Purpose: print a 4 digit number in reverse

#include <stdio.h>

int main()
{   
    // Only allow 4 digit numbers from input to continue
    int num = 0;
    while (num < 1000 || num > 9999)
    {
        printf("Please input 4 digit number: ");
        scanf("%d", &num);
    }

    // Print digits in reverse
    printf("%d in reverse is: ", num);
    while (num != 0)
    {
        printf("%d", num % 10); // Get last digit in number
        num /= 10; // Remove the last digit from number
    }
    printf("\n");
    return 0;
}